src/libs/API/commands/ValidateEmailCommand.js

```diff
@@ -0,0 +1,78 @@
+import lodashGet from 'lodash/get';
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import ONYXKEYS from '../../../ONYXKEYS';
+import * as User from '../../actions/User';
+import BaseAPICommandBlocking from './BaseAPICommandBlocking';
+import CONST from '../../../CONST';
+import Navigation from '../../Navigation/Navigation';
+import ROUTES from '../../../ROUTES';
+
+let isLoggedIn = nul;
+Onyx.connect({
```

Comment:
I find it odd and unintuitive that an Api command class has knowledge about whether the user is logged in, about routes, and that it also does navigation

Reply:
Ah, this is a good point, but I'm not sold on the solution should be that the code remains in actions/User.js. In order to make that happen, this command would have to return a promise (or trigger a callback) so that User.js knows that the data has been processed and inserted into Onyx and now other logic is OK to be done.

I think it's a trade-off of complexity (callbacks and promises) for a more clear separation of logic (onxy management in one place, business logic in another place).

I'm more on the side of keeping all the business logic in the same place as the Onyx data handling, mostly because I haven't seen it be a problem, whereas I have run into problems with excessive callbacks and promises.

Reply:
>this command would have to return a promise (or trigger a callback) so that User.js knows that the data has been processed and inserted into Onyx and now other logic is OK to be done.

Can't User.js subscribe to this data in onyx and react to changes to it? Alternatively, can't User.js use PubSub/DataChanged? 

To clarify and make sure we're talking about the same thing, what specifically do you call business logic?

Reply:
**question:** If we think of this as a "network action" instead of an "API command method" would it feel more intuitive?